### PR TITLE
Make boxes for dask-spark page

### PR DESCRIPTION
**Pull request by martindurant**
Ref : https://github.com/dask/community/issues/2

### CODE REVIEW

**Comment by martindurant** [comment]
Looks like:

Screen Shot 2019-08-07 at 14 25 34

The remaining parts of the previous page is nested by API, so would need at least better headings.

**Comment by jrbourbeau** [comment]
@martindurant your proposal of replacing the nested structure with "DataFrame API", "Array API", etc. seems like a good way forward

File: docs/source/spark.rst [hunk 1]
```diff
@@ -18,84 +18,140 @@ has fewer features and, instead, is used in conjunction with other libraries,
...
+| Provides no such library.                              | Provides GraphX, a library for graph processing.       |
++--------------------------------------------------------+--------------------------------------------------------+
+
+Custom parallelism:
+
++--------------------------------------------------------+--------------------------------------------------------+
+| **Dask**                                               | **Spark**                                              |
++--------------------------------------------------------+--------------------------------------------------------+
+| Allows you to specify arbitrary task graphs for more   | Generally expects users to compose computations out    |
+| complex and custom systems that are not part of the    | of theirhigh-level primitives (map, reduce, groupby,   |
```
Comment (TomAugspurger) [inline docs/source/spark.rst]:
```suggestion
| complex and custom systems that are not part of the    | of their high-level primitives (map, reduce, groupby,   |
```

**Comment by mrocklin** [comment]
@TomAugspurger @jrbourbeau you both think a bit about style from time to time.  Any thoughts on things that we can do to improve styling or layout here?  This is one of our most viewed pages, so making it look nice is probably worth some effort.

**Comment by jrbourbeau** [comment]
As it stands now, I think the "Dask" and "Spark" table column titles stand out more than the "Language", "Ecosystem", etc. category text. We might consider making the "Dask" and "Spark" titles italicized, instead of bold, and make "Language", etc. smaller section titles. E.g. 

```
Language:
```
becomes
```
Language
^^^^^^^^
```

Which would look like

Screen Shot 2019-08-08 at 10 51 13 AM

I'm curious what others think

File: docs/source/spark.rst [hunk 2]
```diff
@@ -18,84 +18,140 @@ has fewer features and, instead, is used in conjunction with other libraries,
...
+| Provides no such library.                              | Provides GraphX, a library for graph processing.       |
++--------------------------------------------------------+--------------------------------------------------------+
+
+Custom parallelism:
+
++--------------------------------------------------------+--------------------------------------------------------+
+| **Dask**                                               | **Spark**                                              |
++--------------------------------------------------------+--------------------------------------------------------+
+| Allows you to specify arbitrary task graphs for more   | Generally expects users to compose computations out    |
+| complex and custom systems that are not part of the    | of their high-level primitives (map, reduce, groupby,   |
```
Comment (jrbourbeau) [inline docs/source/spark.rst]: I'm getting `WARNING: Malformed table.` for the "Custom parallelism" table when building the docs locally. This should fix it:

```suggestion
| complex and custom systems that are not part of the    | of their high-level primitives (map, reduce, groupby,  |
```

Comment (martindurant) [reply docs/source/spark.rst]: Aha, OK. It really shouldn't have to line up...

**Comment by TomAugspurger** [comment]
Looks good. I think a `vertical-align: top` for the table cells might look a bit nicer.

**Comment by martindurant** [comment]
Note: I cannot figure out how to push the text in the boxes to the top

**Comment by TomAugspurger** [comment]
Hmm that's a bit unfortunate. Not worth worrying about though.

**Comment by jrbourbeau** [comment]
Thanks @martindurant!

**Pull request merged**